tests/components/point/test_config_flow.py

```diff
@@ -0,0 +1,202 @@
+"""Tests for the Point config flow."""
+import asyncio
+import sys
+from unittest.mock import Mock, patch
+
+from homeassistant import data_entry_flow
+from homeassistant.components.point import config_flow, DOMAIN
+
+from tests.common import mock_coro
+
+
+async def test_abort_if_no_implementation_registered(hass):
+    """Test we abort if no implementation is registered."""
+    flow = config_flow.PointFlowHandler()
+    flow.hass = hass
+    result = await flow.async_step_user()
+
+    assert result['type'] == data_entry_flow.RESULT_TYPE_ABORT
+    assert result['reason'] == 'no_flows'
+
+
+async def test_abort_if_already_setup(hass):
+    """Test we abort if Point is already setup."""
+    flow = config_flow.PointFlowHandler()
+    flow.hass = hass
+    with patch.object(hass.config_entries, 'async_entries', return_value=[{}]):
+        result = await flow.async_step_user()
+
+    assert result['type'] == data_entry_flow.RESULT_TYPE_ABORT
+    assert result['reason'] == 'already_setup'
+
+    with patch.object(hass.config_entries, 'async_entries', return_value=[{}]):
+        result = await flow.async_step_import()
+    assert result['type'] == data_entry_flow.RESULT_TYPE_ABORT
+    assert result['reason'] == 'already_setup'
+
+
+async def test_full_flow_implementation(hass):
+    """Test registering an implementation and finishing flow works."""
+    config_flow.register_flow_implementation(hass, DOMAIN, 'Test', 'id',
+                                             'secret')
+    config_flow.register_flow_implementation(hass, 'test-other', 'Test Other',
+                                             None, None)
+    point = Mock()
```

Comment:
We could break out this mock and the mock set up to a pytest fixture function, to easily be able to reuse the mock.